### PR TITLE
ENH: Add SimpleITK-documentation target to superbuild

### PR DESCRIPTION
**Pull request by blowekamp**
Add support to passing Documentation and Doxygen related
CMake variables from Superbuild to actual SimpleITK project.

Enable SimpleITK step-level targets.

Add SimpleITK Documentation level step.

### CODE REVIEW

**Comment by blowekamp** [comment]
@fbudin69500

Working on making it easier to build SimpleITK Doxygen.

**Comment by blowekamp** [comment]
@thewtex @fbudin69500 
The nightly doxygen build successes for SimpleITK on CircleCI:
https://circleci.com/gh/SimpleITK/SimpleITK/1380

It includes an artifact for the html for the doxygen. While this patch is now merged, I'd appreciate a review to see how best to move the nightly SimpleITK doxygen to being generated on this platform.

Comments and thoughts appreciated!

File: .circleci/config.yml [hunk 1]
```diff
@@ -80,15 +80,79 @@ jobs:
...
+            sudo pip install --upgrade pip
+            ci_addons circle/install_cmake.py 3.9.2
+      - run:
+          name: Building and Generating Doxygen
+          no_output_timeout: 20.0m
+          environment:
+                CTEST_CONFIGURATION_TYPE: "Release"
+                CTEST_BUILD_FLAGS: "-j 4"
+                CTEST_OUTPUT_ON_FAILURE: 1
+                ITK_GLOBAL_DEFAULT_NUMBER_OF_THREAD: 2
```
Comment (thewtex) [inline .circleci/config.yml]: Why is this set? Also `THREAD` -> `THREADS`

**Comment by thewtex** [comment]
@blowekamp I was able to download the tarball :+1: 

The tarball contains the absolute path `/home/circleci[...]`. It should just have `html/[...]`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -80,15 +80,79 @@ jobs:
...
+          command: |
+            export PATH=/usr/lib/ccache:${PATH}
+            mkdir ${CTEST_BINARY_DIRECTORY} && cd ${CTEST_BINARY_DIRECTORY}
+            cmake -DWRAP_DEFAULT=OFF -DBUILD_DOXYGEN=ON -DSimpleITK_USE_SYSTEM_LUA:BOOL=ON ${CTEST_SOURCE_DIRECTORY}/SuperBuild
+            make -j ${CTEST_BUILD_FLAGS} SimpleITK-doc
+      - run:
+          name: Archiving directory for artifact
+          command: |
+            tar -zcvf simpleitk_doxygen_html.tar.gz ${CTEST_BINARY_DIRECTORY}/SimpleITK-build/Documentation/html
```
Comment (blowekamp) [inline .circleci/config.yml]: To resolve the absolute path in the archive, it should be simply changing the above to:
cd ${CTEST_BINARY_DIRECTORY}/SimpleITK-build/Documentation/
tar -zcvf simpleitk_doxygen_html.tar.gz html

**Pull request merged**